src/main/java/fern/simulation/Simulator.java

```diff
@@ -244,6 +244,15 @@ public void initialize() {
 		for (int i=0; i<a.length; i++) 
 			a[i] = propensityCalculator.calculatePropensity(i, getAmountManager(),this);
```

Comment:
Complete code duplication! Please no copy paste of code, put this  in a helper function `initializePropensities` (plural!) which is called from initialize. Also missing `{` brackets on for loop!
``` 
		for (int i=0; i<a.length; i++) 
			a[i] = propensityCalculator.calculatePropensity(i, getAmountManager(),this);
``